src/hotspot/cpu/riscv/icache_riscv.cpp

```diff
@@ -36,13 +39,20 @@ static int icache_flush(address addr, int lines, int magic) {
   //
   // No sush assurance is defined at the interface level of the builtin
   // method, and so we should make sure it works.
-  __asm__ volatile("fence rw, rw" : : : "memory");
```

Comment:
I think we should keep this inline assembly code as explained by the preceding code comment.
If I remembered correctly, such a data fence is not specified at the syscall interface level.
```
__asm__ volatile("fence rw, rw" : : : "memory");
```
Also you might want to update code comment to reflect the new code.

Reply:
Yes, ok, I'll add it back and fix comment.